pkg/cdi/container-edits.go

```diff
@@ -205,6 +205,22 @@ func (e *ContainerEdits) isEmpty() bool {
 	return len(e.Env)+len(e.DeviceNodes)+len(e.Hooks)+len(e.Mounts) == 0
 }
 
+// minSpecVersion determines the minimum required version for the container edits.
+// This checks whether version-specific fields have been specified and returns the spec version as a semver where these were added.
+func (e *ContainerEdits) minSpecVersion() string {
+	for _, dn := range e.DeviceNodes {
+		if dn.HostPath != "" && dn.HostPath != dn.Path {
+			return "v0.5.0"
+		}
+	}
+	for _, m := range e.Mounts {
+		if m.Type != "" {
+			return "v0.4.0"
+		}
+	}
+	return "v0.2.0"
+}
```

Comment:
That's not really spec related, but dependent on the package version that is doing the injection. (although we did relax that requirement with the `v0.5.0` release, so we could use a spec version to indicate it).

Reply:
I'm just worried that someone generates a spec with devices beginning with numbers, then runs this function to see what minimum version they can set it to. It might tell them they have a min version of v0.2.0 (because the fields they use are all consistent with a minimum version of v0.2.0), but then their device names will throw errors on consumers that are pinned to v0.2.0 because names beginning with numbers weren't possible until v0.5.0.

Reply:
@klueska I have updated the check to include this.